lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,41 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          secrets = nil
+          if config.is_a?(String)
+            secrets = [config]
+          else
+            secrets = config.values
+          end
+
+          tokens = {}
+          secrets.each do |secret|
+            key, secret = secret.split('=').map(&:strip)
+            pull_request = self.data.pull_request ? self.data.pull_request : ""
+            now = Time.now.to_i()
+            payload = {"iss" => "travis-ci.org",
+                      "slug" => self.data.slug,
+                      "pull-request" => pull_request,
+                      "exp" => now+5400,
+                      "iat" => now}
+            token = JWT.encode(payload, secret)
```

Comment:
Okay, i'll commit my code that handles an exception and the test, but I
won't know that is the exception till its tested again.

On Tue, Jul 26, 2016 at 3:57 PM, Hiro Asari <notifications@github.com>
wrote:

> In lib/travis/build/addons/jwt.rb
> <https://github.com/travis-ci/travis-build/pull/356#discussion_r72353621>:
>
> > +            secrets = [config]
> > +          else
> > +            secrets = config.values
> > +          end
> > +
> > +          tokens = {}
> > +          secrets.each do |secret|
> > +            key, secret = secret.split('=').map(&:strip)
> > +            pull_request = self.data.pull_request ? self.data.pull_request : ""
> > +            now = Time.now.to_i()
> > +            payload = {"iss" => "travis-ci.org",
> > +                      "slug" => self.data.slug,
> > +                      "pull-request" => pull_request,
> > +                      "exp" => now+5400,
> > +                      "iat" => now}
> > +            token = JWT.encode(payload, secret)
>
> It's quite possible that my initial analysis was wrong. An exception does
> occur here in some cases, though, so it is important that you handle it.
> I'll take another look later.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/travis-ci/travis-build/pull/356/files/bb16a55b414a22dd66ab41c924251d85a665efff#r72353621>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuB-6BsuvCuHge3d8nO5nrfxs92j4eks5qZpDqgaJpZM4DHeYE>
> .
>